### PR TITLE
Image editing

### PR DESCRIPTION
**Pull request by michaelbromley**
Implements editing of images in nodes.

### CODE REVIEW

File: src/app/editor/form-generator/components/binary-field/binary-field.component.ts [hunk 1]
```diff
@@ -59,30 +76,133 @@ export class BinaryFieldComponent extends BaseFieldComponent {
...
+    }
+
+    editImage(): void {
+        const node = this.api.getNodeValue() as MeshNode;
+        let imageUrl: string | SafeUrl;
+        const newFile = this.binaryProperties.file;
+        if (newFile) {
+            imageUrl = this.blobService.createObjectURL(this.binaryProperties.file);
+        } else {
+            imageUrl = this.apiService.project.getBinaryFileUrl(node.project.name, node.uuid, this.api.field.name);
```
Comment (virbas) [inline src/app/editor/form-generator/components/binary-field/binary-field.component.ts]: modify getBinaryFileURL to accept node.version

File: src/app/editor/form-generator/components/binary-field/binary-field.component.ts [hunk 2]
```diff
@@ -59,30 +76,133 @@ export class BinaryFieldComponent extends BaseFieldComponent {
...
+                height: round(params.cropRect.height * ratio)
+            },
+            focalPointX: params.focalPointX,
+            focalPointY: params.focalPointY
+        };
+    }
+
+    private getBinaryUrl(binaryField: BinaryField): string {
+        const node = this.api.getNodeValue() as MeshNode;
+        let binaryUrl = this.apiService.project.getBinaryFileUrl(node.project.name, node.uuid, this.api.field.name);
```
Comment (virbas) [inline src/app/editor/form-generator/components/binary-field/binary-field.component.ts]: same with node.version parameter

File: src/app/editor/providers/editor-effects.service.spec.ts [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+    });
+
+
+});
+
+class MockEntitiesService {}
+class MockI18nNotification {
+    show = jasmine.createSpy('show');
+}
+class MockConfigService implements ConfigService {
```
Comment (virbas) [inline src/app/editor/providers/editor-effects.service.spec.ts]: Create a global MockConfigService

Comment (michaelbromley) [reply src/app/editor/providers/editor-effects.service.spec.ts]: Done

File: src/app/editor/form-generator/components/binary-field/binary-field.component.ts (hunk 1 above)
Comment (michaelbromley) [reply src/app/editor/form-generator/components/binary-field/binary-field.component.ts]: done

File: src/app/editor/form-generator/components/binary-field/binary-field.component.ts (hunk 2 above)
Comment (michaelbromley) [reply src/app/editor/form-generator/components/binary-field/binary-field.component.ts]: done

**Pull request merged**